### ISSUE TITLE
Provide a way to do async sleep in Guzzle

### ISSUE THREAD

**Issue by mikethea1** [guzzle/guzzle#3263]
**Description**
I'm interacting with an API where you have to poll it while waiting for a result to be ready. Obviously I can poll by looping with PHP's `sleep`/`usleep` functions, but this will block other async work that I have running in parallel from executing during the sleep. What I'd like is something akin to C#'s `Task.Delay()` which lets me do an asynchronous sleep.

I notice that `CurlMultiHandler` has this functionality internally with delays, but there is no way for me to add my own delay without being tied to a request. So today I can emulate this like so:

```
function sleepAsync(float $delaySeconds) {
    return $this->client->getAsync('https://some.fake.domain', ['delay' => $delaySeconds])->then(fn() => null, fn() => null);
}
```
My concern about this is that it still has to go through DNS resolution which is probably more heavy-weight than would be ideal.

One (disruptive) approach would be to push the delay functionality down into the Guzzle async queue abstraction (vs. having it in the CurlMultiHandler), but I could also imagine an API like `CurlMultiHandler->delay($seconds)` that would give you a way to plug in.

I'd also be interested in hearing about alternative approaches to achieving this with the current functionality.

**Example**
```
// using Guzzle's coroutine yielding for async flow
while (true) {
    $response = yield $this->pollAsync($id); // calls Guzzle's getAsync() under the hood
    if ($response->done) { return $response->result; }
    yield $this->sleepAsync(0.01);
}
```

**Comment by mikethea1** [comment]
I see this has been marked as stale, but I'd love to discuss if one of the maintainers has bandwidth to engage

**Comment by mikethea1** [comment]
Is this actually completed or just close - won't fix?
